zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/UserTaskHandler.java

```diff
@@ -0,0 +1,235 @@
+              .setState(TaskState.CREATED);
+      default -> {}
+    }
+
+    final TaskJoinRelationship joinRelation = new TaskJoinRelationship();
+    joinRelation.setName(TaskJoinRelationshipType.TASK.getType());
+    joinRelation.setParent(Long.valueOf(entity.getProcessInstanceId()));
+    entity.setJoin(joinRelation);
+  }
+
+  @Override
+  public void flush(final TaskEntity entity, final BatchRequest batchRequest) {
+
+    final Map<String, Object> updateFields = getUpdatedFields(entity);
+
+    batchRequest.upsertWithRouting(
+        indexName, entity.getId(), entity, updateFields, entity.getProcessInstanceId());
+  }
+
+  private Map<String, Object> getUpdatedFields(final TaskEntity entity) {
+
+    final Map<String, Object> updateFields = new HashMap<>();
+    if (entity.getChangedAttributes() != null) {
+      for (final String changedAttribute : entity.getChangedAttributes()) {
+        switch (changedAttribute) {
+          case "assignee" -> updateFields.put(TaskTemplate.ASSIGNEE, entity.getAssignee());
+          case "candidateGroupsList" ->
+              updateFields.put(TaskTemplate.CANDIDATE_GROUPS, entity.getCandidateGroups());
+          case "candidateUsersList" ->
+              updateFields.put(TaskTemplate.CANDIDATE_USERS, entity.getCandidateUsers());
+          case "dueDate" -> updateFields.put(TaskTemplate.DUE_DATE, entity.getDueDate());
+          case "followUpDate" ->
+              updateFields.put(TaskTemplate.FOLLOW_UP_DATE, entity.getFollowUpDate());
+          case "priority" -> updateFields.put(TaskTemplate.PRIORITY, entity.getPriority());
+          default ->
+              LOGGER.warn(
+                  "Attribute update not mapped while importing ZEEBE_USER_TASKS: {}",
+                  changedAttribute);
+        }
+      }
+    }
+    if (entity.getCompletionTime() != null) {
+      updateFields.put(TaskTemplate.COMPLETION_TIME, entity.getCompletionTime());
+    }
+
+    if (entity.getChangedAttributes() != null && !entity.getChangedAttributes().isEmpty()) {
+      updateFields.put(TaskTemplate.CHANGED_ATTRIBUTES, entity.getChangedAttributes());
+    }
+    if (entity.getAction() != null) {
+      updateFields.put(TaskTemplate.ACTION, entity.getAction());
+    }
+    if (entity.getState() != null) {
+      updateFields.put(TaskTemplate.STATE, entity.getState());
+    }
+    if (entity.getFlowNodeBpmnId() != null) {
+      updateFields.put(TaskTemplate.FLOW_NODE_BPMN_ID, entity.getFlowNodeBpmnId());
+    }
+    if (entity.getProcessDefinitionId() != null) {
+      updateFields.put(TaskTemplate.PROCESS_DEFINITION_ID, entity.getProcessDefinitionId());
+    }
+    if (entity.getBpmnProcessId() != null) {
+      updateFields.put(TaskTemplate.BPMN_PROCESS_ID, entity.getBpmnProcessId());
+    }
+
+    return updateFields;
+  }
+
+  private void createTaskEntity(final TaskEntity entity, final Record<UserTaskRecordValue> record) {
```

Comment:
❓ Is the join will be included on the creation of Process Entity? (I may find the answer while I am reviewing)

Reply:
The Join for the TaskEntity is being set on the `updateEntity`. The same applies for the rest of the Task related handles